src/cls/rgw/cls_rgw.cc

```diff
@@ -1090,6 +1194,11 @@ int rgw_bucket_complete_op(cls_method_context_t hctx, bufferlist *in, bufferlist
                        __func__, escape_str(idx).c_str(), rc);
           return rc;
         }
+
+        bufferlist empty;
+        rc = record_duplicate_entry(hctx, idx, entry.key, &empty, header.resharding_in_logrecord());
+        if (rc < 0)
+          return rc;
```

Comment:
can replace this and the `cls_cxx_map_remove_key()` call just above with `remove_entry()`